src/nvim/shada.c

```diff
@@ -3315,16 +3315,16 @@ static ShaDaReadResult msgpack_read_uint64(ShaDaReadDef *const sd_reader, const
   RERR "Error while reading ShaDa file: " \
   entry_name " entry at position %" PRIu64 " " \
   error_desc
-#define CHECK_KEY(key, expected) ( \
-                                   key.via.str.size == sizeof(expected) - 1 \
-                                   && STRNCMP(key.via.str.ptr, expected, sizeof(expected) - 1) == 0)
+#define CHECK_KEY(key, expected) \
+  ((key).via.str.size == (sizeof(expected) - 1) \
+   && STRNCMP((key).via.str.ptr, expected, (sizeof(expected) - 1)) == 0)  // NOLINT
```

Comment:
@zeertzjq, what do you of this? 
I could never get clint happy, and if I use the complete rule name then the line is longer than 100 characters, and I can't disable a category either :(

It's basically https://github.com/neovim/neovim/issues/17763

Reply:
Best way to make clint happy is to join as many lines as possible and then run uncrustify. This passes clint:


```c
#define CHECK_KEY(key, \
                  expected) ((key).via.str.size == (sizeof(expected) - 1) \
                             && STRNCMP((key).via.str.ptr, expected, (sizeof(expected) - 1)) == 0)
```